File: src/base/presenter/UsrPresenter.java
package base.presenter;

import base.model.Usr;

import java.util.ArrayList;

/**
 * Created by MurphySL on 2017/12/13 10:08
 */
public class UsrPresenter extends Presenter<Usr> {

    @Override
    public void insert(Usr usr) {
        jni.insertUsr(usr);
    }

    @Override
    public void delete(Usr usr) {
        jni.deleteUsr(usr);
    }

    @Override
    public int select(Usr usr) {
        return jni.selectUsr(usr);
    }

    @Override
    public ArrayList<Usr> selectAll() {
        return jni.selectAllUsr();
    }

    @Override
    public void update(Usr usr) {
        jni.updateUsr(usr);
    }
}

File: src/base/presenter/UsrPresenter.kt
package base.presenter

import base.model.Usr

/**
 *
 * Created by MurphySL on 2017/12/13 19:31
 */
class UsrPresenter : Presenter<Usr>() {

    override fun insert(t: Usr): Int = 1

    override fun delete(t: Usr) = Unit

    override fun update(t: Usr) : Int = 1

    override fun select(t: Usr): Int = 1

    fun selectAll() : ArrayList<Usr> = ArrayList()

}